dasmith.rb

```diff
@@ -18,47 +17,28 @@ class DASmith < Sinatra::Base
     require 'newrelic_rpm'
   end
 
-  configure do
-
-    # Pull all of the article-images into @images array
-    @images = []
-    Dir[File.dirname(__FILE__) + '/public/article-images/*.{png,jpg,jpeg,gif,webp,svg}'].each { |img| @images << img }
-
-    # Pull in the markdown files and put them in the $articles array.
-    puts 'Loading Articles...'
-    Dir[File.dirname(__FILE__) + '/articles/*.md'].each do |f|
-      article = Article.new f, @images
-      puts File.basename f
-      $articles << article
-    end
 
-    # Load Draft Articles if we're in development
-    if ENV['RACK_ENV'] == 'development'
-      puts 'Loading Draft Articles...'
-      Dir[File.dirname(__FILE__) + '/articles/drafts/*.md'].each do |f|
-        article = Article.new f, @images
-        puts File.basename f
-        $articles << article
-      end
-    end
-
-    # Sort in publish date order
-    $articles.sort! { |a, b| a.publish_date <=> b.publish_date }
-    $articles.reverse!
+  configure do
 
     # Set up Readit config
     Readit::Config.consumer_key = ENV['READABILITY_KEY']
     Readit::Config.consumer_secret = ENV['READABILITY_SECRET']
     Readit::Config.parser_token = ENV['READABILITY_PARSER_TOKEN']
+
+    # Read in Articles
+    Article.configure do |config|
+      config.articles_path = 'articles'
+      config.draft_articles_path = 'articles/drafts'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.